### ISSUE TITLE
domain reliability patch is obsolete / superfluous

### ISSUE THREAD

**Issue by thestinger** [bromite/bromite#986]
In `chrome/browser/domain_reliability/service_factory.cc` there are already conditions for enabling the service. It would be a lot simpler to change `const bool kDefaultEnabled = true;` to `const bool kDefaultEnabled = false;` than the invasive changes. However, it's enabled disabled if `ChromeMetricsServiceAccessor::IsMetricsAndCrashReportingEnabled` is `false`. Even if it was force enabled, it also already discards the results by default if metrics are disabled.

Maybe there's some other reason you want to have this but it would be simpler to drop the patch. I think it would be helpful to at least document that it should be entirely redundant.

**Comment by csagan5** [comment]
This is a patch from the ungoogled-chromium project, and it appears to be still there as well: https://github.com/Eloston/ungoogled-chromium/blob/master/patches/core/ungoogled-chromium/disable-domain-reliability.patch

I reckon it could be simplified with something like:
```diff
diff --git a/chrome/browser/domain_reliability/service_factory.cc b/chrome/browser/domain_reliability/service_factory.cc
index b65f4bf78589..eb6260bfa13a 100644
--- a/chrome/browser/domain_reliability/service_factory.cc
+++ b/chrome/browser/domain_reliability/service_factory.cc
@@ -27,18 +27,7 @@ const char DomainReliabilityServiceFactory::kUploadReporterString[] = "chrome";
 
 // static
 bool DomainReliabilityServiceFactory::ShouldCreateService() {
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  if (command_line->HasSwitch(switches::kDisableDomainReliability))
-    return false;
-  if (command_line->HasSwitch(switches::kEnableDomainReliability))
-    return true;
-  if (!ChromeMetricsServiceAccessor::IsMetricsAndCrashReportingEnabled())
-    return false;
-  if (base::FieldTrialList::TrialExists(kFieldTrialName)) {
-    std::string value = base::FieldTrialList::FindFullName(kFieldTrialName);
-    return (value == kFieldTrialValueEnable);
-  }
-  return kDefaultEnabled;
+  return false;
 }
 
 }  // namespace domain_reliability
```

@Eloston you might be interested in this too

**Comment by csagan5** [comment]
We have opted to keep the current patch but also add the above change. If it is cumbersome to maintain we might want to simplify the patch further as suggested here; thanks again for reporting this!

Fixed in `90.0.4430.59`.
